components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -368,6 +369,44 @@ public MLModel retrieveModel(long modelId) throws MLModelBuilderException {
         }
     }
 
+    /**
+     * Publish a ML model to registry.
+     *
+     * @param tenantId    Unique ID of the tenant.
+     * @param userName    Username of the user.
+     * @param modelId     Unique ID of the built ML model.
+     * @throws MLModelBuilderException
+     */
+    public void publishModel(int tenantId, String userName, long modelId) throws MLModelBuilderException {
+        InputStream in = null;
+        try {
+            // read model
+            MLStorage storage = databaseService.getModelStorage(modelId);
+            String storageType = storage.getType();
+            String storageLocation = storage.getLocation();
+            MLIOFactory ioFactory = new MLIOFactory(mlProperties);
+            MLInputAdapter inputAdapter = ioFactory.getInputAdapter(storageType + MLConstants.IN_SUFFIX);
+            in = inputAdapter.read(new URI(storageLocation));
+            // create registry path
+            MLCoreServiceValueHolder valueHolder = MLCoreServiceValueHolder.getInstance();
+            String modelName = databaseService.getModel(tenantId, userName, modelId).getName();
```

Comment:
This could lead to a NPE. Better to split it to two instructions and do a null check.